src/main/java/com/puppycrawl/tools/checkstyle/JavaParser.java

```diff
@@ -136,7 +136,7 @@ public static DetailAST parseFileText(FileText text, Options options)
     public static DetailAST parseFile(File file, Options options)
             throws IOException, CheckstyleException {
         final FileText text = new FileText(file.getAbsoluteFile(),
-            System.getProperty("file.encoding", StandardCharsets.UTF_8.name()));
+            StandardCharsets.UTF_8.name());
```

Comment:
this is interesting .... why JavaParser does not use charset defined by user in Checker ?

Reply:
Unless I'm mistaken, `parseFile()` is only called when printing the AST, right?  It's never taking in the config file anyway.

Reply:
you are right, I was confused by code and location of it.

it might be good sign that methods location is not very good.